File: ConexionBD/Conexion.py
from enum import Enum
import mysql.connector

class TipoMotor(Enum):
    MYSQL = "MYSQL"
    SQLSERVER = "SQLSERVER"

class TipoAutenticacion(Enum):
    CREDENCIALES = "CREDENCIALES"
    WINDOWS = "WINDOWS"

class Conexion:
    _motorActual = TipoMotor.MYSQL
    _autenticacion = TipoAutenticacion.CREDENCIALES
    _host = "localhost"
    _puertoInstancia = "3306"
    _baseDatos = "corporativo_pos"
    _usuario = "root"
    _contrasena = "2306"
    _instancia = None

    @staticmethod
    def getMotor():
        return Conexion._motorActual

    @staticmethod
    def configurar(motor, auth, h, pi, bd, u, pass_):
        Conexion._motorActual = motor
        Conexion._autenticacion = auth
        Conexion._host = h
        Conexion._puertoInstancia = pi
        Conexion._baseDatos = bd
        Conexion._usuario = u
        Conexion._contrasena = pass_
        Conexion._instancia = None

    @staticmethod
    def getConexion():
        try:
            if Conexion._instancia is None or not Conexion._instancia.is_connected():
                if Conexion._motorActual == TipoMotor.MYSQL:
                    Conexion._instancia = mysql.connector.connect(
                        host=Conexion._host,
                        port=Conexion._puertoInstancia,
                        database=Conexion._baseDatos,
                        user=Conexion._usuario,
                        password=Conexion._contrasena
                    )
                    print("Conexion con MySQL establecida")
                else:
                    print("SQL Server no implementado en este script")
                    pass
                Conexion.asegurarColumnas()
        except Exception as e:
            print(e)
        return Conexion._instancia

    @staticmethod
    def cerrar():
        try:
            if Conexion._instancia is not None and Conexion._instancia.is_connected():
                Conexion._instancia.close()
                print("Conexion cerrada")
        except Exception as e:
            print(e)

    @staticmethod
    def asegurarColumnas():
        try:
            if Conexion._instancia is not None and Conexion._instancia.is_connected():
                cursor = Conexion._instancia.cursor()
                if Conexion._motorActual == TipoMotor.MYSQL:
                    sql = "ALTER TABLE ventas ADD COLUMN descuento DECIMAL(10,2) DEFAULT 0 AFTER total"
                else:
                    sql = "ALTER TABLE ventas ADD descuento DECIMAL(10,2) DEFAULT 0"
                cursor.execute(sql)
                Conexion._instancia.commit()
                cursor.close()
                print("Columna 'descuento' asegurada en la base de datos.")
        except Exception:
            pass

File: ConexionBD/Conexion.cs
using MySqlConnector;
using System;

namespace punto_de_venta_C_.ConexionBD
{
    public class Conexion
    {
        public enum TipoMotor
        {
            MYSQL, SQLSERVER
        }

        public enum TipoAutenticacion
        {
            CREDENCIALES, WINDOWS
        }

        private static TipoMotor motorActual = TipoMotor.MYSQL;
        private static TipoAutenticacion autenticacion = TipoAutenticacion.CREDENCIALES;

        private static string host = "localhost";
        private static string puertoInstancia = "3306";
        private static string baseDatos = "corporativo_pos";
        private static string usuario = "root";
        private static string contrasena = "2306";

        private static MySqlConnection instancia = null;

        public static TipoMotor getMotor()
        {
            return motorActual;
        }

        public static void configurar(TipoMotor motor, TipoAutenticacion auth, string h, string pi, string bd, string u, string pass)
        {
            motorActual = motor;
            autenticacion = auth;
            host = h;
            puertoInstancia = pi;
            baseDatos = bd;
            usuario = u;
            contrasena = pass;
            instancia = null;
        }

        public static MySqlConnection getConexion()
        {
            try
            {
                if (instancia == null || instancia.State == System.Data.ConnectionState.Closed)
                {
                    string url = $"Server={host};Port={puertoInstancia};Database={baseDatos};Uid={usuario};Pwd={contrasena};AllowPublicKeyRetrieval=true;";
                    instancia = new MySqlConnection(url);
                    instancia.Open();
                    Console.WriteLine("Conexion con MySQL establecida");
                    asegurarColumnas();
                }
            }
            catch (Exception e)
            {
                Console.Error.WriteLine("Error de conexion: " + e.Message);
            }
            return instancia;
        }

        public static void cerrar()
        {
            try
            {
                if (instancia != null && instancia.State != System.Data.ConnectionState.Closed)
                {
                    instancia.Close();
                    Console.WriteLine("Conexion cerrada");
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }

        private Conexion() { }

        private static void asegurarColumnas()
        {
            try
            {
                if (instancia != null && instancia.State != System.Data.ConnectionState.Closed)
                {
                    string sql = "ALTER TABLE ventas ADD COLUMN descuento DECIMAL(10,2) DEFAULT 0 AFTER total";
                    using (MySqlCommand st = new MySqlCommand(sql, instancia))
                    {
                        st.ExecuteNonQuery();
                    }
                    Console.WriteLine("Columna 'descuento' asegurada en la base de datos.");
                }
            }
            catch (Exception) { }
        }
    }
}
